config/yaml-mp/src/main/java/io/helidon/config/yaml/mp/YamlMpConfigSource.java

```diff
@@ -196,10 +196,25 @@ public static List<ConfigSource> classPath(String resource) {
      * @return list of config sources discovered (may be zero length)
      */
     public static List<ConfigSource> classPath(String resource, String profile) {
+        return classPath(resource, profile, null);
+    }
+
+    /**
+     * Create from YAML file(s) on classpath from specified classloader with profile support.
+     *
+     * @param resource resource name to locate on classpath (looks for all instances)
+     * @param profile configuration profile to use
+     * @param classLoader classloader where resource will be retrieved from (will use the
+     *                    current thread's classloader if value is {@code null})
+     * @return list of config sources discovered (may be zero length)
+     */
+    public static List<ConfigSource> classPath(String resource, String profile, ClassLoader classLoader) {
```

Comment:
Once again - we cannot have a public API method that accepts nulls

Reply:
implementation now validates profile and classloader must not be null